src/compiler/factory.ts

```diff
@@ -1595,6 +1606,321 @@ namespace ts {
         return createVoid(createLiteral(0));
     }
 
+    export function createExportDefault(expression: Expression) {
+        return createExportAssignment(/*decorators*/ undefined, /*modifiers*/ undefined, /*isExportEquals*/ false, expression);
+    }
+
+    export function createExternalModuleExport(exportName: Identifier) {
+        return createExportDeclaration(/*decorators*/ undefined, /*modifiers*/ undefined, createNamedExports([createExportSpecifier(exportName)]));
+    }
+
+    // Utilities
+
+    function asName(name: string): Identifier;
+    function asName(name: Identifier): Identifier;
+    function asName(name: string | Identifier): Identifier;
+    function asName(name: string | BindingName): BindingName;
+    function asName(name: string | PropertyName): PropertyName;
+    function asName(name: string | Identifier | BindingName | PropertyName) {
+        return typeof name === "string" ? createIdentifier(name) : name;
+    }
+
+    function asExpression(value: string | number | Expression) {
+        return typeof value === "string" || typeof value === "number" ? createLiteral(value) : value;
+    }
+
+    function asNodeArray<T extends Node>(array: T[] | undefined): NodeArray<T> | undefined {
+        return array ? createNodeArray(array) : undefined;
+    }
+
+    function asToken<TKind extends SyntaxKind>(value: TKind | Token<TKind>): Token<TKind> {
+        return typeof value === "number" ? createToken(value) : value;
+    }
+
+    /**
+     * Clears any EmitNode entries from parse-tree nodes.
+     * @param sourceFile A source file.
+     */
+    export function disposeEmitNodes(sourceFile: SourceFile) {
+        // During transformation we may need to annotate a parse tree node with transient
+        // transformation properties. As parse tree nodes live longer than transformation
+        // nodes, we need to make sure we reclaim any memory allocated for custom ranges
+        // from these nodes to ensure we do not hold onto entire subtrees just for position
+        // information. We also need to reset these nodes to a pre-transformation state
+        // for incremental parsing scenarios so that we do not impact later emit.
+        sourceFile = getSourceFileOfNode(getParseTreeNode(sourceFile));
+        const emitNode = sourceFile && sourceFile.emitNode;
+        const annotatedNodes = emitNode && emitNode.annotatedNodes;
+        if (annotatedNodes) {
+            for (const node of annotatedNodes) {
+                node.emitNode = undefined;
+            }
+        }
+    }
+
+    /**
+     * Associates a node with the current transformation, initializing
+     * various transient transformation properties.
+     *
+     * @param node The node.

```

Comment:
These `@param node The node`s are low value (as are a lot of the others), so I'd delete them.